### PR TITLE
User routes backend

### PR DESCRIPTION
**Pull request by tiongMax**
**What i did:**

- Setup user route
- Create user route handler
- Get user route handler
- Get all user route handler
- Delete user route handler
- Update user route handler
- Fix login and logout issue
- Combine user and blacklist database into 1.
- Test all of the route handlers above using POSTMAN

### CODE REVIEW

File: user-service/src/controllers/userController.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import User from '../models/userModel';
+import generateTokenAndSetCookie from '../lib/generateToken';
+import mongoose from "mongoose";
+
+export async function createUser(req: Request, res: Response) {
+  try {
+    const { username, email, password } = req.body;
+
+    // sanity check
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (zacwong2151) [inline user-service/src/controllers/userController.ts:12]: can leave a comment to show what this regex means?

File: user-service/src/controllers/userController.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		if (existingUser) {
+			return res.status(400).json({ error: "Username is already taken" });
+		}
+
+		const existingEmail = await User.findOne({ email });
+		if (existingEmail) {
+			return res.status(400).json({ error: "Email is already taken" });
+		}
+
+		if (password.length < 10) {
```
Comment (zacwong2151) [inline user-service/src/controllers/userController.ts:27]: maybe can reduce it so that we can easily put short passwords for testing purposes 🥲

**Review by zacwong2151 — COMMENTED** [review]
some comments

File: user-service/src/controllers/userController.ts (hunk 2 above)
Comment (brein62) [reply user-service/src/controllers/userController.ts:27]: i mean does it truly matter when the frontend password validation is even stricter :')

File: user-service/src/controllers/userController.ts (hunk 1 above)
Comment (brein62) [reply user-service/src/controllers/userController.ts:12]: image

hope this helps @tiongMax

Comment (tiongMax) [reply user-service/src/controllers/userController.ts:14]: Ok

File: user-service/src/controllers/userController.ts (hunk 2 above)
Comment (tiongMax) [reply user-service/src/controllers/userController.ts:29]: Ok

Comment (zacwong2151) [reply user-service/src/controllers/userController.ts:29]: we have frontend password validation?

Comment (zacwong2151) [reply user-service/src/controllers/userController.ts:29]: why wd i want to write `admin12345` why i can write `asdf` :) all for efficiency

Comment (brein62) [reply user-service/src/controllers/userController.ts:29]: IMG_6206

oof ow owie

File: user-service/src/controllers/userController.ts (hunk 1 above)
Comment (tiongMax) [reply user-service/src/controllers/userController.ts:15]: ok done.

File: user-service/src/controllers/userController.ts (hunk 2 above)
Comment (zacwong2151) [reply user-service/src/controllers/userController.ts:30]: oof this is my first time seeing the login page ngl

**Review by wjayee — APPROVED** [review]
LGTM!

**Pull request merged**